src/main/java/com/ms3_inc/camel/extensions/rest/SwaggerRequestValidator.java

```diff
@@ -16,91 +16,143 @@
+
+        openapi4jValidator = new RequestValidator(api);
+    }
+
+    public SwaggerRequestValidator(String specPath, String basePath) {
+        api = null;
+
+        try {
+            api = new OpenApi3Parser()
+                    .parse(new ClassPathResource(specPath).getFile(), false);
+        } catch (ResolutionException re) {
+            LOGGER.error(re.getMessage());
+        } catch (ValidationException ve) {
+            LOGGER.error(ve.getMessage());
+        } catch (IOException ioe) {
+            LOGGER.error(ioe.getMessage());
+        }
+
+        api.setServers(new ArrayList<Server>(){
+            {
+                add(0, new Server().setUrl(basePath));
+            }
+        });
+
+        openapi4jValidator = new RequestValidator(api);
     }
 
     @Override
     public boolean process(Exchange exchange, AsyncCallback callback) {
         LOGGER.debug("Trying validation");
 
         try {
-            ValidationReport report = validator.validateRequest(fromExchange(exchange));
-            if (report.hasErrors()) {
-                LOGGER.debug(report.toString());
-                exchange.setException(new BadRequestException(fromReport(report)));
-            }
-
-            LOGGER.debug("Validating complete");
-        } catch (Exception ex) {
-            exchange.setException(new CamelException(ex));
+            openapi4jValidator.validate(fromExchange(exchange));
+        } catch (ValidationException e) {
+            ValidationResults results = e.results();
+            LOGGER.debug(e.results().toString());
+            exchange.setException(new BadRequestException(fromReport(results)));
         } finally {
+            LOGGER.debug("Validating complete");
             callback.done(true);
         }
 
         return true;
     }
 
-    private static OperationResult fromReport(ValidationReport report) {
+    private static OperationResult fromReport(ValidationResults results) {
         List<OperationResult.Message> answer = new ArrayList<>(1);
+        StringBuilder resultItems = new StringBuilder();
 
-        // separates the report into a "Validation failed." line for details and the rest for diagnostics
-        String[] result = SimpleValidationReportFormat.getInstance().apply(report).split(System.lineSeparator(), 2);
-        answer.add(new OperationResult.Message(OperationResult.Level.ERROR, null, result[0], result[1]));
+        // separates the results into a "Validation failed." line for details and the rest for diagnostics
+        for (ValidationResults.ValidationItem result : results.items()) {
+            resultItems.append(result.message() + "\n");
+        }
+        answer.add(new OperationResult.Message(OperationResult.Level.ERROR, null, "Validation failed.", resultItems.toString()));
 
         return new OperationResult(answer);
     }
 
     private static Request fromExchange(Exchange exchange) {
+        final Logger LOGGER = LoggerFactory.getLogger(SwaggerRequestValidator.class);
```

Comment:
Don't we already have a LOGGER declared in the class?

Reply:
Yes, good point, I just need to make the first LOGGER static.